generate/schemesgen/main.go

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2017, Shreyas Khare <skhare@rapid7.com>
+// See LICENSE for licensing information
+
+package main
+
+import (
+	"net/http"
+	"io/ioutil"
+	"io"
+	"text/template"
+	"os"
+	"log"
+	"encoding/csv"
+	"strings"
+)
+
+const path = "standard_schemes.go"
+
+var schemesTmpl = template.Must(template.New("schemes").Parse(`// Generated by schemesgen
+
+package xurls
+
+// Schemes is a sorted list of all IANA assigned schemes.
+//
+// Source:
+//   https://www.iana.org/assignments/uri-schemes/uri-schemes-1.csv
+var StdSchemes = []string{
+{{range $scheme := .Schemes}}` + "\t`" + `{{$scheme}}` + "`" + `,
+{{end}}}
+`))
+
+func schemeList() []string {
+	resp, _ := http.Get("https://www.iana.org/assignments/uri-schemes/uri-schemes-1.csv")
+	file, _ := ioutil.ReadAll(resp.Body)
+	r := csv.NewReader(strings.NewReader(string(file)))
```

Comment:
these three lines should instead be:

```
resp, err := http.Get("https://www.iana.org/assignments/uri-schemes/uri-schemes-1.csv")
if err != nil {
    log.Fatal(err)
}
defer resp.Body.Close()
csv.NewReader(resp.Body)
```

Reply:
Fixed. I'm really liking the `defer` statement!